frontend/src/component/admin/users/UsersList/ResetPassword/ResetPassword.tsx

```diff
@@ -0,0 +1,123 @@
+import React, { useState } from 'react';
+import classnames from 'classnames';
+import { Box, styled, TextField, Typography } from '@mui/material';
+import { modalStyles } from 'component/admin/users/util';
+import { Dialogue } from 'component/common/Dialogue/Dialogue';
+import { useThemeStyles } from 'themes/themeStyles';
+import { IUser } from 'interfaces/user';
+import useAdminUsersApi from 'hooks/api/actions/useAdminUsersApi/useAdminUsersApi';
+import { UserAvatar } from 'component/common/UserAvatar/UserAvatar';
+import useToast from 'hooks/useToast';
+import { formatUnknownError } from 'utils/formatUnknownError';
+import { LinkField } from '../../LinkField/LinkField';
+
+const StyledUserAvatar = styled(UserAvatar)(({ theme }) => ({
+    width: theme.spacing(5),
+    height: theme.spacing(5),
+    margin: 0,
+}));
+
+interface IChangePasswordProps {
+    showDialog: boolean;
+    closeDialog: () => void;
+    user: IUser;
+}
+
+const ResetPassword = ({
+    showDialog,
+    closeDialog,
+    user,
+}: IChangePasswordProps) => {
+    const { classes: themeStyles } = useThemeStyles();
+    const { resetPassword } = useAdminUsersApi();
+    const { setToastApiError } = useToast();
+    const [resetLink, setResetLink] = useState('');
+
+    const submit = async (event: React.SyntheticEvent) => {
+        event.preventDefault();
+        if (!user.email) {
+            setToastApiError(
+                "You can't reset the password of a user who doesn't have an email address.",
+            );
+            return;
+        }
+
+        try {
+            const token = await resetPassword(user.email).then((res) =>
+                res.ok ? res.json() : undefined,
+            );
+
+            if (token) {
+                setResetLink(token.resetPasswordUrl);
+            } else {
+                setToastApiError(
+                    'Could not reset password. If you made another request in less than a minute, wait for one minute and try again.',
```

Comment:
Maybe something like this?

```suggestion
                    'Could not reset password. This may be to prevent too many resets. Try again in a minute.',
```